### ISSUE TITLE
`torch.nn.HuberLoss` backwards unexpectedly fail

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#76646]
### 🐛 Describe the bug

`torch.nn.HuberLoss` backwards unexpectedly fail with error message `Found dtype Float but expected Double`. 
For its documentation, "When delta is set to 1, this loss is equivalent to [SmoothL1Loss](https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html#torch.nn.SmoothL1Loss)". As a result, it should succeed to backward like `SmoothL1Loss`

```python
import torch
t1_tensor = torch.rand([67, 1], dtype=torch.float64)
t2_tensor = torch.rand([67, 1], dtype=torch.float32)

t1 = t1_tensor.clone().detach().requires_grad_().to('cpu')
t2 = t2_tensor.clone().detach().requires_grad_().to('cpu')
HL = torch.nn.HuberLoss()
try:
    HL(t1, t2).sum().backward()
except Exception as e:
    print("HuberLoss Fail Backward:", e)

t1 = t1_tensor.clone().detach().requires_grad_().to('cpu')
t2 = t2_tensor.clone().detach().requires_grad_().to('cpu')
SL = torch.nn.SmoothL1Loss()
SL(t1, t2).sum().backward()
print("SmoothL1Loss Succeed Backward")

# HuberLoss Fail Backward: Found dtype Float but expected Double
# SmoothL1Loss Succeed Backward
```

### Versions

pytorch: 1.11.0

**Comment by qwertyforce** [comment]
hi! i've encountered the same issue on the latest torch, could you make this actionable? i've prepared a small fix similar to how it was fixed in mse_loss (https://github.com/pytorch/pytorch/pull/175743)

**Comment by soulitzer** [comment]
We're happy to accept a PR that allows backward of huber loss also type promote properly
